.mage/proto.go

```diff
@@ -0,0 +1,95 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"context"
+	"github.com/magefile/mage/mg"
+	"go.einride.tech/mage-tools/mglog"
+	"go.einride.tech/mage-tools/mgpath"
+	"go.einride.tech/mage-tools/mgtool"
+	"go.einride.tech/mage-tools/tools/mgbuf"
+	"go.einride.tech/mage-tools/tools/mgclangformat"
+)
+
+type Proto mg.Namespace
+
+func (Proto) All() {
+	mg.Deps(
+		Proto.ClangFormatProto,
+		Proto.BufLint,
+		Proto.BufGenerate,
+	)
+	mg.Deps(
+		Proto.BufGenerateExample,
+	)
+}
+
+func (Proto) BufLint(ctx context.Context) error {
+	mglog.Logger("buf-lint").Info("linting protos...")
+	cmd := mgbuf.Command(ctx, "lint")
+	cmd.Dir = mgpath.FromGitRoot("proto")
+	return cmd.Run()
+}
+
+func (Proto) ClangFormatProto() error {
+	mglog.Logger("clang-format-proto").Info("formatting protos...")
+	cmd := mgclangformat.FormatProtoCommand(mgpath.FromGitRoot("proto"))
+	return cmd.Run()
+}
+
+func (Proto) ProtocGenGo(ctx context.Context) error {
+	_, err := mgtool.GoInstallWithModfile(
+		ctx,
+		"google.golang.org/protobuf/cmd/protoc-gen-go",
+		mgpath.FromGitRoot("go.mod"),
+	)
+	return err
+}
+
+func (Proto) ProtocGenGoGRPC(ctx context.Context) error {
+	_, err := mgtool.GoInstall(ctx, "google.golang.org/grpc/cmd/protoc-gen-go-grpc", "v1.2.0")
+	return err
+}
+
+func (Proto) ProtocGenGoCLI() error {
+	mglog.Logger("protoc-gen-go-aip").Info("building binary...")
+	return mgtool.Command(
+		"go",
+		"build",
+		"-o",
+		mgpath.FromBins("protoc-gen-go-cli"),
+		mgpath.FromGitRoot("."),
+	).Run()
```

Comment:
Should this not run in the proto dir?

Reply:
It doesn't need to - in fact, it's building the binary from the main file in the parent directory